libjolie/src/main/java/jolie/lang/parse/module/SymbolReferenceResolver.java

```diff
@@ -567,6 +567,9 @@ public void visit( TypeDefinitionLink n ) {
 					return;
 				}
 				linkedType = (TypeDefinition) targetSymbolInfo.get().node();
+				while( linkedType instanceof TypeDefinitionLink ) {
```

Comment:
Don't we risk that this loops infinitely in cases like the following? If so, we might fix this by putting an upper bound (100?).

```
type A:B
type B:A
```